gokart/mypy.py

```diff
@@ -73,10 +72,44 @@ def get_base_class_hook(self, fullname: str) -> Callable[[ClassDefContext], None
                 return self._task_on_kart_class_maker_callback
         return None
 
+    def get_function_hook(self, fullname: str) -> Callable[[FunctionContext], Type] | None:
+        """Adjust the return type of the `Field` function."""
+        if PARAMETER_FULLNAME_MATCHER.match(fullname):
+            return self._task_on_kart_parameter_field_callback
+        return None
+
     def _task_on_kart_class_maker_callback(self, ctx: ClassDefContext) -> None:
         transformer = TaskOnKartTransformer(ctx.cls, ctx.reason, ctx.api)
         transformer.transform()
 
+    def _task_on_kart_parameter_field_callback(self, ctx: FunctionContext) -> Type:
+        """Extract the type of the `default` argument from the Field function, and use it as the return type.
+
+        In particular:
+        * Retrieve the type of the argument which is specified, and use it as return type for the function.
+        * If no default argument is specified, return AnyType with unannotated type instead of parameter types like `luigi.Parameter()`
+          This makes mypy avoid conflict between the type annotation and the parameter type.
+          e.g.
+          ```python
+          foo: int = luigi.IntParameter()
+          ```
+        """
+
+        default_args = ctx.args[0]
+
+        if default_args:
+            default_type = ctx.arg_types[0][0]
+            default_arg = default_args[0]
+
+            # Fallback to default Any type if the field is required
+            if not isinstance(default_arg, EllipsisExpr):
+                return default_type
+        # NOTE: This is a workaround to avoid the error between type annotation and parameter type.
+        #       As the following code snippet, the type of `foo` is `int` but the assigned value is `luigi.IntParameter()`.
+        #       foo: int = luigi.IntParameter()
+        # TODO: infer mypy type from the parameter type.
+        return AnyType(TypeOfAny.unannotated)
```

Comment:
If no default value, a parameter handled as `AnyType(TypeOfAny.unannotated)`
Due to this, type annotation for the `TaskOnKart` takes precedence.